pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -152,7 +152,7 @@ func convertEndpoint(service *model.Service, servicePort *networking.Port,
 			},
 			LbWeight: endpoint.Weight,
 			Labels:   endpoint.Labels,
-			TLSMode:  tlsMode,
+			TLSMode:  model.GetTLSModeFromEndpointLabels(endpoint.Labels, endpoint.ServiceAccount),
```

Comment:
Not it is not any more from just labels. Can we rename this to `GetTLSModeFromEndpoint`  or just call `GetTLSMode` ?

Reply:
sounds good